src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+
+### Key Principles of REST
+1. **Stateless**: Every client request to the server must include all necessary information for comprehension and processing. The server should not retain any data about the client's latest HTTP request.
+
+2. **Client-Server Architecture**: The client handles the user interface and experience, while the server manages request processing and resource management.
+
+3. **Cacheable**: Clients can cache responses from the server, enhancing performance by reducing the server load and minimizing client-server interactions.
+
+4. **Uniform Interface**: The API has a consistent interface, which simplifies the architecture and enhances visibility of interactions.
+
+## REST API Architecture
+REST APIs utilize a client-server architecture, communicating over HTTP. The client sends requests to the server, which processes them and sends back responses. Communication is stateless, requiring each client request to contain all necessary information for processing.
+
+![REST API](/src/assets/RestApi/images/rest_api.png)
+
+### Components of REST API Architecture
+- **Resources**: In REST, resources are identified by URLs, representing anything that can be named and addressed, such as documents, images, temporal services, collections of other resources, non-virtual objects, and more.
+
+- **HTTP Methods**: REST APIs utilize standard HTTP methods for operations on resources. Common methods include GET (to retrieve a resource), POST (to create a new resource), PUT (to update a resource), and DELETE (to remove a resource).
+
+- **Representations**: Resources are represented in a format easily understood by both client and server, commonly JSON (JavaScript Object Notation) or XML (eXtensible Markup Language).
+
+### Implementing REST API
+Now, lets start to build a REST API. Before we start, make sure you have Python installed on your system. You can download Python from [python.org](https://www.python.org/). We will also need Flask, which you can install using pip:
+```bash
+pip install Flask
+```
+
+We will build a REST API for managing a list of tasks. Each task will have a title and a status indicating whether it's completed or not.
+
+1. Create a new Python file: `app.py`
+
+2. Import Flask and initialize the app:
+```python
+from flask import Flask, jsonify, request
+
+app = Flask(__name__)
+```
+
+3. Define the data structure for our tasks. For simplicity, we'll use a list of dictionaries:
+```python
+tasks = [
+    {"id": 1, "title": "Buy groceries", "completed": False},
+    {"id": 2, "title": "Clean the house", "completed": True}
+]
+```
+
+### GET Requests
+To retrieve all tasks, implement a route that handles GET requests:
```

Comment:
```suggestion
- To retrieve all tasks, implement a route that handles GET requests:
- To retrieve a specific task
```

it would be better to clarify the distinction between the two methods for the GET request by summarizing them as separate points or other way you prefer? This would enhance clarity